File: Dao/pedidos-react-23-main/pedidos-react-23-main/src/services/articuloService.js
import api from "./pedidosAPI";

class ArticuloService {
  buscarTodos() {
    return api.get("/articulos");
  }

  buscarPorId(id) {
    return api.get(`/articulos/${id}`);
  }

  crear(data) {
    return api.post("/articulos", data);
  }

  modificar(id, data) {
    return api.put(`/articulos/${id}`, data);
  }

  eliminar(id) {
    return api.delete(`/articulos/${id}`);
  }

  buscarPorDescripcion(descripcion) {
    return api.get(`/articulos?descripcion=${descripcion}`);
  }

  buscarPorFamiliaId(familiaId) {
    return api.get(`/articulos?familiaId=${familiaId}`);
  }
}

const service = new ArticuloService();
export default service;

File: Dao/pedidos-rest-24-main/pedidos-rest-24-main/src/main/java/es/uvigo/mei/pedidos/servicios/ArticuloService.java
package es.uvigo.mei.pedidos.servicios;

import java.util.List;
import java.util.Optional;

import es.uvigo.mei.pedidos.entidades.Articulo;
import es.uvigo.mei.pedidos.entidades.Familia;

public interface ArticuloService {
	public Articulo crear(Articulo articulo);
	public Articulo modificar(Articulo articulo);
	public void eliminar(Articulo articulo);
	public Optional<Articulo> buscarPorId(Long id);
	public List<Articulo> buscarTodos();
	public List<Articulo> buscarPorDescripcion(String patron);
	public List<Articulo> buscarPorFamilia(Long idFamilia);
	
	public Familia crearFamilia(Familia familia);
	public Familia modificarFamilia(Familia familia);
	public void eliminarFamilia(Familia familia);
	public Optional<Familia> buscarFamiliaPorId(Long id);
	public List<Familia> buscarFamilias();
	public List<Familia> buscarFamiliasPorDescripcion(String patron);
	
}
